recline.theme.inc

```diff
@@ -0,0 +1,546 @@
+<?php
+
+/**
+ * @file
+ * Recline theme functions.
+ */
+define('MAX_SIZE_PREVIEW_ZIP', 600000);
+
+/**
+ * Returns HTML for an recline field formatter.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - item: Associative array of recline field
+ *
+ * @ingroup themeable
+ */
+function theme_recline_default_formatter($vars) {
+  $type = null;
+
+  if (isset($vars['item']['entity']->field_format['und'][0]['tid'])) {
+    $format = taxonomy_term_load($vars['item']['entity']->field_format['und'][0]['tid']);
+    $type = $format->name;
+  }
+
+  if (isset($vars['item']['url'])) {
+    $url = $vars['item']['url'];
+  }
+  elseif (isset($vars['item']['uri'])) {
+    $url = file_create_url($vars['item']['uri']);
+  }
+
+  if($url) {
+    $type = (!$type || $type == 'data') ? recline_get_data_type(recline_remote_mime($url)) : $type;        
+  }
+
+  $default_file = array(
+    'filemime' => $type,
+    'filesize' => '',
+    'filename' => $url,
+    'description' => '',
+  );
+  $file = isset($vars['item']['fid']) ? $vars['item'] : $default_file;
+  if (is_object($file)) {
+    $file = (array) $file;
+  }
+  $description = isset($file['description']) ? $file['description'] : '';
+  $output = recline_build_icon($url, $type, $file['filename'], $file['filemime'], $file['filesize'], $description);
+  $request = drupal_http_request($url, array('timeout' => 2, 'method' => 'HEAD'));
+
+  if ($request->code != 200) {
+    if ($request->code != 405 && $request->code != 400) {
+      $output['preview'] = recline_preview_unavailable();
+      return drupal_render($output);
+    }
+    else {
+      $request = drupal_http_request($url, array('timeout' => 10));
+      if ($request->code != 200) {
+        $output['preview'] = recline_preview_unavailable();
+        return drupal_render($output);
+      }
+    }
+  }
+  switch ($type) {
+    case 'json':
+      $output['preview'] = recline_preview_json($url);
+      break;
+    case 'csv':
```

Comment:
Add xls type for this. and also switch the function name to recline_preview_multiview.